src/Core/Security/Domain/ProviderConfiguration/OpenId/Model/ActiveConfiguration.php

```diff
@@ -0,0 +1,121 @@
+
+declare(strict_types=1);
+
+namespace Core\Security\Domain\ProviderConfiguration\OpenId\Model;
+
+use Core\Contact\Domain\Model\ContactGroup;
+use Core\Contact\Domain\Model\ContactTemplate;
+use Centreon\Domain\Common\Assertion\Assertion;
+use Core\Security\Domain\ProviderConfiguration\OpenId\Exceptions\OpenIdConfigurationException;
+
+/**
+ * This class represent an active OpenId Configuration
+ *
+ * active configuration should have mandatory parameters to be able to authenticate using this configuration.
+ */
+class ActiveConfiguration extends AbstractConfiguration
+{
+    protected bool $isActive = true;
+
+    /**
+     * @param boolean $isAutoImportEnabled
+     * @param string|null $clientId
+     * @param string|null $clientSecret
+     * @param string|null $baseUrl
+     * @param string|null $authorizationEndpoint
+     * @param string|null $tokenEndpoint
+     * @param string|null $introspectionTokenEndpoint
+     * @param string|null $userInformationEndpoint
+     * @param ContactGroup|null $contactGroup
+     * @param ContactTemplate|null $contactTemplate
+     * @param string|null $emailBindAttribute
+     * @param string|null $userAliasBindAttribute
+     * @param string|null $userNameBindAttribute
+     * @throws OpenIdConfigurationException
+     */
+    public function __construct(
+        protected bool $isAutoImportEnabled,
+        protected ?string $clientId = null,
+        protected ?string $clientSecret = null,
+        protected ?string $baseUrl = null,
+        protected ?string $authorizationEndpoint = null,
+        protected ?string $tokenEndpoint = null,
+        protected ?string $introspectionTokenEndpoint = null,
+        protected ?string $userInformationEndpoint = null,
+        protected ?ContactGroup $contactGroup = null,
+        protected ?ContactTemplate $contactTemplate = null,
+        protected ?string $emailBindAttribute = null,
+        protected ?string $userAliasBindAttribute = null,
+        protected ?string $userNameBindAttribute = null,
+    ) {
+        $this->validateMandatoryParameters();
+    }
+
+    /**
+     * Validate Mandatory Parameters
+     *
+     * @throws OpenIdConfigurationException
+     */
+    private function validateMandatoryParameters(): void
+    {
+        Assertion::notEmpty($this->clientId, "Configuration::clientId");
+        Assertion::notEmpty($this->clientSecret, "Configuration::clientSecret");
+        Assertion::notEmpty($this->baseUrl, "Configuration::baseUrl");
+        Assertion::notEmpty($this->authorizationEndpoint, "Configuration::authorizationEndpoint");
+        Assertion::notEmpty($this->tokenEndpoint, "Configuration::tokenEndpoint");
+        Assertion::notNull($this->contactGroup, "Configuration::contactGroup");
+        if (empty($this->introspectionTokenEndpoint) && empty($this->userInformationEndpoint)) {
+            throw OpenIdConfigurationException::missingInformationEndpoint();
+        }
+        $this->validateParametersForAutoImport();
```

Comment:
Move those controls outside of models and create builder that will handle business logic and validation, in Application